### PR TITLE
write all the docs

### PR DESCRIPTION
**Pull request by quarterto**

### CODE REVIEW

File: tools/x-docs/plugins/gatsby-plugin-doc-pages/gatsby-node.js [hunk 1]
```diff
@@ -97,6 +122,24 @@ exports.createPages = ({ boundActionCreators, graphql }) => {
...
+				const filePaths = nodes.map(
+					({fileAbsolutePath}) => '  ◆ ' + chalk.cyan(path.relative(repoBase, fileAbsolutePath))
+				).join('\n');
+
+				console.warn();
+				console.warn(`${chalk.black.keyword('black').bold.bgYellow(' WARNING ')} there are multiple files that would have the url ${chalk.blue.underline(url)}:
+
+${filePaths}
+
+which file is chosen is arbitrary. you should move of the files to another folder.
```
Comment (i-like-robots) [inline tools/x-docs/plugins/gatsby-plugin-doc-pages/gatsby-node.js]: Maybe reword this to something like:

"This will cause unexpected output, please re-name or move the files to resolve the conflict"

File: tools/x-docs/src/docs/guides/apps/setup.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+The templates are authored as abstract [JSX](https://reactjs.org/docs/introducing-jsx.html), an extension of Javascript that allows you to use HTML-like markup as a first-class concept (that is, you can define it, pass it around and use it like strings or objects). Since you can use variables in JSX expressions, we can use them to provide templating functionality, similar to Handlebars.
+
+These components do not contain any state or behaviour, so they don't require any kind of runtime to work. Because of this, they can be used in any React-like library, such as Preact or Inferno, without relying on the React runtime, or without any runtime at all, outputting static HTML strings. If you're using a library like one of these, x-dash components will work with [minimal setup](#setup). If you're not using one, and you want to use a component without any setup, they can be used as plain functions that output strings, similar to a Handlebars template. They can also be used as Handlebars template partials by installing [x-handlebars](/packages/x-handlebars).
```
Comment (i-like-robots) [inline tools/x-docs/src/docs/guides/apps/setup.md]: Maybe reword "They can also be used as Handlebars template partials by installing x-handlebars" to something like "They can also be integrated with Handlebars via the x-handlebars module"

File: tools/x-docs/src/docs/guides/apps/setup.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+## Installing
+
+Components are installed with [`npm`](https://npmjs.org). See the individual [component's page](/tools/x-docs/src/docs/components/index.md) for exact instructions, but you'll be running something like:
```
Comment (i-like-robots) [inline tools/x-docs/src/docs/guides/apps/setup.md]: "See the individual component's page for exact instructions, but you'll be running something like"

=> "For example to install the x-teaser component"

File: tools/x-docs/src/docs/guides/apps/setup.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+  image: {
+    url: 'https://example.com/foo.jpg'
+  },
+  showImage: true
+});
+```
+
+### Use component presets
+
+Because enabling toggles can become repetetive, and to make use cases clearer, some components export presets.
```
Comment (i-like-robots) [inline tools/x-docs/src/docs/guides/apps/setup.md]: speeling

File: tools/x-docs/src/docs/guides/apps/setup.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+The Teaser component has a `Small` preset, enabling the "title", "meta" and "status" elements (which show title, concept tags and live blog status & article timestamp respectively):
```
Comment (i-like-robots) [inline tools/x-docs/src/docs/guides/apps/setup.md]: remove live blog mention?

File: tools/x-docs/src/docs/guides/components/conventions.md [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+## Syntax
+
+x-dash components are made up of functions that return JSX elements. These are known as [Functional Components](https://reactjs.org/docs/components-and-props.html#functional-and-class-components) in React.
+
+Each component should be in a separate file, with the name of the component in `PascalCase` as the filename and `.jsx` as the extension, for example `LiveBlogStatus.jsx`.
+
+The `main` entry point of the component package should be the main component file itself, which should be named in the same format as all component files, instead of `index.js` or similar. It should export the main component as a named export, not `default`.
+
+If a component is made up of many sub-components, it should export the sub-components used to assemble the main component as named exports alongside the main component.
```
Comment (i-like-robots) [inline tools/x-docs/src/docs/guides/components/conventions.md]: "it should export all sub-components in addition to the main component"

File: tools/x-docs/src/docs/guides/components/cookbook/composing-components.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
+# Composing components
+
+Some components have slots that allow consumers to add extra elements in certain places. These can be other components or strings of HTML. One example of this is the [Teaser component's "actions" slot](https://github.com/financial-times/x-dash/tree/master/components/x-teaser/src/CustomSlot.jsx), which is typically used for a topic follow button.
```
Comment (i-like-robots) [inline tools/x-docs/src/docs/guides/components/cookbook/composing-components.md]: It's just called `customSlot` now

Comment (i-like-robots) [reply tools/x-docs/src/docs/guides/components/cookbook/composing-components.md]: Which still makes me gag a bit

File: tools/x-docs/src/docs/guides/components/cookbook/composing-components.md [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+## Using a component's custom slot
+
+In the component's documentation will be a list of its properties and what it expects. In the Teaser component's case, its property `customSlot` will be rendered in an `o-teaser__action` element. This can be a JSX element or an HTML string.
```
Comment (i-like-robots) [inline tools/x-docs/src/docs/guides/components/cookbook/composing-components.md]: Turns out the `__action` bit is actually specific to one app so I should re-name it 🤦‍♂️

File: tools/x-docs/src/docs/guides/components/getting-started.md [hunk 9]
```diff
@@ -0,0 +1,40 @@
+## Very quick start
+
+1. Clone the x-dash monorepo
+
+    ```bash
+    git clone git@github.com:financial-times/x-dash
```
Comment (i-like-robots) [inline tools/x-docs/src/docs/guides/components/getting-started.md]: `cd x-dash`

File: tools/x-docs/src/docs/guides/components/setup.md [hunk 10]
```diff
@@ -0,0 +1,105 @@
+## Working on components
+
+You've been through the [quick start guide](/tools/x-docs/src/docs/guides/components/getting-started.md), and you want to make your first change to x-dash. This guide will take you through the process for working on x-dash components, demoing and testing your changes, and getting them merged and released.
+
+## Component development process
+
+Open the monorepo in your editor and [have a look around](#getting-around-the-monorepo). While the development server is running, whenever you make a change to a component, it will update in the component explorer without a reload. This way, you can quickly verify your changes in the browser.
```
Comment (i-like-robots) [inline tools/x-docs/src/docs/guides/components/setup.md]: "component explorer" = workbench?

File: tools/x-docs/src/docs/guides/components/setup.md [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+
+x-dash's tools are built around a [set of conventions](/tools/x-docs/src/docs/guides/components/conventions.md) that components should follow. While you can write a component that doesn't follow the conventions, keeping to them will make the development process easier, and make the component easier to use for more consumers.
+
+### The review process
+
+When it's time to make a Pull Request for your changes, request a review from the [`financial-times/x-dash` Github team](https://github.com/orgs/Financial-Times/teams/x-dash). Before the Pull Request is merged, two developers must approve it: one from Next, and one from the Apps team.
+
+This process is probably more restrictive than you're used to. x-dash components should work across a wide range of environments, so we need to make sure the changes you're making are compatible.
+
+We encourage you to open the Pull Request as soon as possible so you can get early feedback. You could even open a Pull Request with an empty commit to start the discussion:
```
Comment (i-like-robots) [inline tools/x-docs/src/docs/guides/components/setup.md]: 👍

File: tools/x-docs/src/docs/guides/components/setup.md (hunk 10 above)
Comment (quarterto) [reply tools/x-docs/src/docs/guides/components/setup.md]: yeah. i'm not 100% happy with the terminology here. we should make it consistent everywhere. which might just mean calling it storybook

File: tools/x-docs/src/docs/guides/apps/setup.md [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+
+Then when you import a component, you can use it with React:
+
+```jsx
+const React = require('react');
+const ReactDOM = require('react-dom');
+const {Teaser} = require('@financial-times/x-teaser');
+
+ReactDOM.render(
+  <Teaser title='Hello world' />,
```
[the remark this answers is not in the document]
Comment (quarterto) [reply tools/x-docs/src/docs/guides/apps/setup.md]: I think people who care about JSX know that, and people who don't care don't care?

File: tools/x-docs/src/docs/guides/components/conventions.md (hunk 6 above)
Comment (quarterto) [reply tools/x-docs/src/docs/guides/components/conventions.md]: Teaser currently doesn't export all its sub-components, just the ones used directly in `Teaser.jsx`

File: tools/x-docs/src/docs/guides/components/cookbook/composing-components.md (hunk 7 above)
Comment (quarterto) [reply tools/x-docs/src/docs/guides/components/cookbook/composing-components.md]: might need a hand on the wording for this
